tests/dash/proto_utils.py

```diff
@@ -107,185 +92,129 @@ def parse_dash_proto(key: str, proto_dict: dict):
+            pbi.encap_type = get_enum_type_from_str('EncapType', json_obj.get("encap_type"))
+        if json_obj.get("vni") is not None:
+            pbi.vni = int(json_obj["vni"])
+        pb.items.append(pbi)
     return pb
 
 
-def route_from_json(json_obj):
-    pb = Route()
-    if json_obj["action_type"] == "vnet":
-        pb.action_type = RoutingType.ROUTING_TYPE_VNET
-        pb.vnet = json_obj["vnet"]
-    elif json_obj["action_type"] == "vnet_direct":
-        pb.action_type = RoutingType.ROUTING_TYPE_VNET_DIRECT
-        pb.vnet_direct.vnet = json_obj["vnet"]
-        pb.vnet_direct.overlay_ip.ipv4 = socket.htonl(int(ipaddress.IPv4Address(json_obj["overlay_ip"])))
-    elif json_obj["action_type"] == "direct":
-        pb.action_type = RoutingType.ROUTING_TYPE_DIRECT
-    else:
-        pytest.fail("Unknown action type %s" % json_obj["action_type"])
-    return pb
+def get_message_from_table_name(table_name):
+    table_name_lis = table_name.lower().split("_")
+    table_name_lis2 = [item.capitalize() for item in table_name_lis]
+    message_name = ''.join(table_name_lis2)
+    module_name = f'dash_api.{table_name.lower()}_pb2'
 
+    # Import the module dynamically
+    module = importlib.import_module(module_name)
 
-def route_rule_from_json(json_obj):
-    pb = RouteRule()
-    pb.action_type = RoutingType.ROUTING_TYPE_VNET_ENCAP
-    pb.priority = int(json_obj["priority"])
-    pb.pa_validation = json_obj["pa_validation"] == "true"
-    if json_obj["pa_validation"] == "true":
-        pb.vnet = json_obj["vnet"]
-    return pb
+    # Get the class object
+    message_class = getattr(module, message_name)
 
+    return message_class()
 
-def routing_type_from_json(json_obj):
-    pb = RouteType()
-    pbi = RouteTypeItem()
-    pbi.action_name = json_obj["name"]
-    pbi.action_type = ActionType.ACTION_TYPE_MAPROUTING
-    pb.items.append(pbi)
-    return pb
 
+def prefix_to_ipv4(prefix_length):
+    mask = 2**32 - 2**(32-int(prefix_length))
+    s = str(hex(mask))
+    s = s[2:]
+    hex_groups = [s[i:i+2] for i in range(0, len(s), 2)]
+    ipv4_address_str = '.'.join(hex_groups)
+    return ipv4_address_str
 
-def acl_group_from_json(json_obj):
-    pb = AclGroup()
-    pb.guid.value = bytes.fromhex(uuid.UUID(json_obj["guid"]).hex)
-    pb.ip_version = IpVersion.IP_VERSION_IPV4
-    return pb
 
+def prefix_to_ipv6(prefix_length):
+    mask = 2**128 - 2**(128-int(prefix_length))
+    s = str(hex(mask))
+    s = s[2:]
+    hex_groups = [s[i:i+4] for i in range(0, len(s), 4)]
+    ipv6_address_str = ':'.join(hex_groups)
+    return ipv6_address_str
 
-def acl_out_from_json(json_obj):
-    pb = AclOut()
-    pb.v4_acl_group_id = json_obj["acl_group_id"]
-    return pb
 
+def parse_ip_prefix(ip_prefix_str):
+    ip_addr_str, mask = ip_prefix_str.split("/")
+    if mask.isdigit():
+        ip_addr = ip_address(ip_addr_str)
+        if ip_addr.version == 4:
+            mask_str = prefix_to_ipv4(mask)
```

Comment:
Hi @aronovic , the mask_str is like this "ff.ff.ff.00" when the mask is "24". Looks like it should be "255.255.255.0".